xline/src/storage/kv_store.rs

```diff
@@ -496,6 +498,13 @@ where
 
     /// Handle `RangeRequest`
     fn handle_range_request(&self, req: &RangeRequest) -> Result<RangeResponse, ExecuteError> {
+        req.validation()?;
+        Self::check_revision(
+            &UniRequest::RequestRange(req.clone()),
+            self.compacted_revision(),
+            self.revision(),
+        )?;
```

Comment:
We already validated it in `kv_server`, Is it necessary to validate again here

Reply:
Currently the validation in kv store is responsible for checking request from xline client which use curp to propose. 

For a etcd client it's might a bit redundant for validating twice, but it may improve performance to validate the reuqest in `kv_server` as we can return immediately if validation has failed, instead of sending to `kv_store` in execution stage and then return a failure.